cmake/modules/configuration_files.cmake

```diff
@@ -116,5 +116,5 @@ DTC_OVERLAY_FILE=\"dts1.overlay dts2.overlay\"")
 # The DTC_OVERLAY_FILE variable is now set to its final value.
 zephyr_boilerplate_watch(DTC_OVERLAY_FILE)
 
-zephyr_get(EXTRA_CONF_FILE SYSBUILD LOCAL VAR EXTRA_CONF_FILE OVERLAY_CONFIG)
-zephyr_get(EXTRA_DTC_OVERLAY_FILE SYSBUILD LOCAL)
+zephyr_get(EXTRA_CONF_FILE SYSBUILD LOCAL VAR EXTRA_CONF_FILE OVERLAY_CONFIG MERGE REVERSE)
+zephyr_get(EXTRA_DTC_OVERLAY_FILE SYSBUILD LOCAL MERGE REVERSE)
```

Comment:
> However, for settings where the value last in the list will overwrite
values earlier in the list, we want the list to be reversed, examples
of such can be CONF_FILE, OVERLAY_CONFIG, DTC_OVERLAY_FILE, where the
content of the file last in the list will overrule the content from an
earlier file.

This seems to reverse the inner content from each scope, not just the order of the scopes, which I am pretty sure is not something desirable.

For example, with `-DEXTRA_CONF_FILE=a.conf;b.conf`, you would expect the `b.conf` to be processed after `a.conf`; but, the current implementation does the opposite.

Reply:
good point.
Taking a look.

Reply:
@stephanosio fixed.